shapely/strtree.py

```diff
@@ -77,11 +82,16 @@ def __init__(
         # Keep references to geoms
         self._geometries = np.asarray(geoms, dtype=np.object_)
         # prevent modification
+        self._prev_writeable = self._geometries.flags.writeable
         self._geometries.flags.writeable = False
 
         # initialize GEOS STRtree
         self._tree = lib.STRtree(self.geometries, node_capacity)
 
+    def __del__(self):
+        # release write lock on geometry array
+        self._geometries.flags.writeable = self._prev_writeable
```

Comment:
Should we put this in a try/except to be sure? I have seen other instances where this was done to prevent AttributeError in case the attribute was already deleted before this is called (not sure about the exact circumstances this happens, but in general one shouldn't rely on `__del__` always being called at the moment you expect)

Reply:
Just so I follow, do you mean something like this?

```python
def __del__(self):
    try:
        self._geometries.flags.writeable = self._prev_writeable
    except:
        pass
```

So that it doesn't break if either `self._geometries` or `self._prev_writeable` has already been deleted?

Reply:
Yes